lib/chef/knife/vsphere_vm_nic_add.rb

```diff
@@ -0,0 +1,49 @@
+require 'chef/knife'
+require 'chef/knife/base_vsphere_command'
+require 'rbvmomi'
+require 'netaddr'
+
+class Chef::Knife::VsphereVmNicAdd < Chef::Knife::BaseVsphereCommand
+  banner 'knife vsphere vm nic add VMNAME NETWORKNAME'
+
+  common_options
+
+  def run
+    $stdout.sync = true
+    vmname = @name_args[0]
+    if vmname.nil?
+      show_usage
+      fatal_exit('You must specify a virtual machine name')
+    end
+
+    networkname = @name_args[1]
+    if networkname.nil?
+      show_usage
+      fatal_exit('You must specify the network name')
+    end
+
+    vim_connection
+
+    dc = datacenter
+    folder = find_folder(get_config(:folder)) || dc.vmFolder
+
+    vm = traverse_folders_for_vm(folder, vmname) || abort("VM #{vmname} not found")
+
+    backing = RbVmomi::VIM.VirtualEthernetCardNetworkBackingInfo(deviceName: networkname)
```

Comment:
I think this might have problems on distributed switches. I just added a few nics and they're not connecting. They look ok in the vcenter gui. If I set them to another network and back again, it works.

I believe you need to give it a port, not a name:

```
    switch, pg_key = network.collect 'config.distributedVirtualSwitch', 'key'
 port = RbVmomi::VIM.DistributedVirtualSwitchPortConnection(
   switchUuid: switch.uuid,
    portgroupKey: pg_key)
  summary = network.name
  backing = RbVmomi::VIM.VirtualEthernetCardDistributedVirtualPortBackingInfo(port: port)
```